src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -297,6 +297,18 @@ private async Task SignalEntityAsyncInternal(TaskHubClient client, string hubNam
                 isReplay: false);
         }
 
+        private bool ConnectionNameMatchesCurrentApp(DurableClient client)
```

Comment:
We can technically be even pickier, by verifying that this DurableClient is also targeting the same task hub that the current application is targeting.

Reply:
Good point!